From 6c2aae4f5dae05d12b834ea8ca5c7da505ffd965 Mon Sep 17 00:00:00 2001
From: Antonio de Angelis <Antonio.deAngelis@arm.com>
Date: Thu, 18 Sep 2025 11:17:46 +0100
Subject: [PATCH 1/4] CC3XX: Add logging on cc3xx_dcu.c

Helps understanding which values are being applied and the
current status of the system (current DCU opens, DCU locks and
the restriction mask).

Upstream-Status: Backport [7d3931b4f02ea253f065d593743a7c2e0cbca0d7]
Signed-off-by: Antonio de Angelis <antonio.deangelis@arm.com>
Change-Id: I426ee064a0008d8031aabdea91fa771b8c892fe4

---
 .../cc3xx/low_level_driver/src/cc3xx_dcu.c    | 29 +++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/platform/ext/target/arm/drivers/cc3xx/low_level_driver/src/cc3xx_dcu.c b/platform/ext/target/arm/drivers/cc3xx/low_level_driver/src/cc3xx_dcu.c
index bc23ed6aba..ce9b1afc4a 100644
--- a/platform/ext/target/arm/drivers/cc3xx/low_level_driver/src/cc3xx_dcu.c
+++ b/platform/ext/target/arm/drivers/cc3xx/low_level_driver/src/cc3xx_dcu.c
@@ -10,6 +10,9 @@
 #include <assert.h>
 #include <string.h>
 
+/* FixMe: Remove this when CC3XX_INFO logging gets sorted */
+#define CC3XX_INFO(...)
+
 /**
  * @brief Check that the requested permissions are in accordance with the
  *        hardware restriction mask
@@ -21,6 +24,12 @@ static cc3xx_err_t check_dcu_restriction_mask(const uint32_t *val)
 {
     size_t idx;
 
+    CC3XX_INFO("icv_dcu_restriction_mask: 0x%08x_%08x_%08x_%08x\r\n",
+               P_CC3XX->ao.ao_icv_dcu_restriction_mask[0],
+               P_CC3XX->ao.ao_icv_dcu_restriction_mask[1],
+               P_CC3XX->ao.ao_icv_dcu_restriction_mask[2],
+               P_CC3XX->ao.ao_icv_dcu_restriction_mask[3]);
+
     for (idx = 0; idx < sizeof(P_CC3XX->ao.ao_icv_dcu_restriction_mask) / sizeof(uint32_t); idx++) {
         if (val[idx] & ~P_CC3XX->ao.ao_icv_dcu_restriction_mask[idx]) {
             return CC3XX_ERR_DCU_MASK_MISMATCH;
@@ -42,6 +51,18 @@ static cc3xx_err_t check_dcu_locks(const uint32_t *val)
     size_t idx;
     uint32_t dcu_has_to_change;
 
+    CC3XX_INFO("Current host_dcu_en: 0x%08x_%08x_%08x_%08x\r\n",
+               P_CC3XX->ao.host_dcu_en[0],
+               P_CC3XX->ao.host_dcu_en[1],
+               P_CC3XX->ao.host_dcu_en[2],
+               P_CC3XX->ao.host_dcu_en[3]);
+
+    CC3XX_INFO("host_dcu_lock: 0x%08x_%08x_%08x_%08x\r\n",
+               P_CC3XX->ao.host_dcu_lock[0],
+               P_CC3XX->ao.host_dcu_lock[1],
+               P_CC3XX->ao.host_dcu_lock[2],
+               P_CC3XX->ao.host_dcu_lock[3]);
+
     for (idx = 0; idx < sizeof(P_CC3XX->ao.host_dcu_en) / sizeof(uint32_t); idx++) {
         /* Check if the host_dcu_en has to change */
         dcu_has_to_change = P_CC3XX->ao.host_dcu_en[idx] ^ val[idx];
@@ -123,6 +144,12 @@ cc3xx_err_t cc3xx_dcu_set_enabled(const uint8_t *permissions_mask, size_t len)
         dcu_en_requested[idx] = *((uint32_t *)(permissions_mask + (idx*sizeof(uint32_t))));
     }
 
+    CC3XX_INFO("Requested host_dcu_en: 0x%08x_%08x_%08x_%08x\r\n",
+               dcu_en_requested[0],
+               dcu_en_requested[1],
+               dcu_en_requested[2],
+               dcu_en_requested[3]);
+
     /* Check the restriction mask for the dcu_en*/
     err = check_dcu_restriction_mask(dcu_en_requested);
     if (err != CC3XX_ERR_SUCCESS) {
@@ -139,6 +166,8 @@ cc3xx_err_t cc3xx_dcu_set_enabled(const uint8_t *permissions_mask, size_t len)
         P_CC3XX->ao.host_dcu_en[idx] = dcu_en_requested[idx];
     }
 
+    CC3XX_INFO("Requested host_dcu_en applied successfully\r\n");
+
     return CC3XX_ERR_SUCCESS;
 }
 /** @} */ // end of cc3xx_dcu
-- 
2.43.0

